### ISSUE TITLE
Enable CI for Windows OS (appveyor?)

### ISSUE THREAD

**Issue by ek9** [ApiGen/ApiGen#831]
It would be nice to have tests run on Windows OS so we can make sure compatibility with Windows OS. I see that there are a few issues regarding path handling that makes tests fail for windows users from time to time.

What do you guys think? I also only know of appveyour, so I suggest trying that out.

**Comment by ek9** [comment]
Related to #804 #824

**Comment by TomasVotruba** [comment]
My experience with AppVeyor is bitter sweet. I've tried it, it helped me fix few bugs on Windows, but I had to also modify few passing tests on Linux and sometimes duplicated them, one for Windows, one for Linux. Windows was broken for no fixable reason so I removed it after 3 months.

If you use Windows and are able to add it, fix it and keep it running, let's roll :+1:

**Comment by ek9** [comment]
I haven't had much experience with it, but it seems helpful for keeping things compatible with windows (and sorting out a few of the windows related issues we are now having). I'm willing to give it a try and see how it goes. If it ever reaches the point where we need to duplicate the tests, then we need to investigate, as the tests should be OS-agnostic. There are various tools that can help with that.

**Comment by ek9** [comment]
We need an owner or someone with admin rights to enable Appveyor account. I've tried doing it, but it still put repository under ek9/appveyor name. Otherwise, this is merged into `4.2` now and will reach master soon.

**Comment by TomasVotruba** [comment]
Could you send me link? I might try it, but probably @kukulich is needed

**Comment by ek9** [comment]
https://ci.appveyor.com/projects/new (I just signup by using GitHub Oauth). It asks to request access to the ApiGen repository (which I did). But I believe only if owners sets it up, it will be shown as `apigen/apigen`, otherwise it uses `username/apigen`.

**Comment by TomasVotruba** [comment]
It's ok, feel free to use your link.

It always uses the username: https://ci.appveyor.com/project/TomasVotruba/autoserviceregistration for github.com/Symplify/AutoServiceRegistration, which I own.

**Comment by ek9** [comment]
Ok, then I guess we do not need further setup. I just need to figure out why it does not start CI builds automatically and I have to start every new build manually. Perhaps it needs to reach master branch first.

**Comment by ek9** [comment]
The tests seem to run automatically if PR is to `master` branch. I will close this issue once I can merge #837 is merged.

**Comment by ek9** [comment]
Closing as it's on master now and seems to be running automatically on PRs.

**Comment by jankal** [comment]
@TomasVotruba can the other contributors somehow also get access to the appveyor build-log?

Everytime I click a link with 'details' abeout the build, Appveyor tells me to be unauthorized to view the log. Even if I'm logged in with my GitHub account in Appveyor.

**Comment by ek9** [comment]
Can you check now? It should be public.

**Comment by jankal** [comment]
Nope

![appveyor](https://cloud.githubusercontent.com/assets/6324332/24723275/355b1b14-1a47-11e7-8f18-847aed85b217.PNG)

**Comment by ek9** [comment]
Which URL are you using? I tried it by using private browsing mode and it opened: https://ci.appveyor.com/project/ek9/apigen

There is no `apigen/apigen` project as apparently that's how appveyor handles this (it does not properly do organization projects).

**Comment by jankal** [comment]
@ek9

Your URL is working.

So nobody is able to just click the link of the integration here in GitHub besides @TomasVotruba cause that's the one linked everywhere (mainly pulls). Pretty complicated for newbies I think.

**Comment by ek9** [comment]
Apparently appveyor suggests to make an organization (separate) account for that: https://www.appveyor.com/docs/team-setup/#setting-up-appveyor-account-for-github-organization
If someone can manage to setup account for `apigen` - let me know, I can update the badges, otherwise all the settings are in `appveyor.yml` so it will pick up the changes automatically on the new account.

**Comment by jankal** [comment]
@TomasVotruba or @kukulich could you please do that cause there needs to be someone trustworthy who has the credentials for that account then? I consider you two cause you both are 'contributors from the early beginning'.

**Comment by kukulich** [comment]
I'm working on it.

**Comment by kukulich** [comment]
https://ci.appveyor.com/project/apigen/apigen

@ek9 @jankal @TomasVotruba You need to register to AppVeyor and send me your emaill address so I can add you as collaborators.

**Comment by jankal** [comment]
@kukulich `himself@alexanderjank.de` - will delete here afterwards.

**Comment by kukulich** [comment]
@jankal I find new settings in AppVeyor based on GitHub Teams. Does it work for you?

**Comment by jankal** [comment]
@kukulich Yes, as I see, I now have access but I'm still not able to do specific things like creating a new build.

**Comment by kukulich** [comment]
@jankal I've just tried it in another way. Is it better?

**Comment by jankal** [comment]
@kukulich Yes. Now it works like I think it should.
